src/transports/ws/sws.c

```diff
@@ -773,14 +744,21 @@ static int nn_sws_fail_conn (struct nn_sws *self, int code, char *reason)
 
     self->fail_msg_len += payload_len;
 
-    self->instate = NN_SWS_INSTATE_FAILING;
+    self->instate = NN_SWS_INSTATE_CLOSED;
 
-    /*  On the next recv, the connection will be failed. Why defer
-        until the next recv? Semantically, until then, this incoming
-        message has not been interpreted, so it's not until then that
-        it could be failed. This type of pre-processing is necessary
-        to early fail chunked transfers. */
```

Comment:
@gdamore -- I notice your change here "more sanely" immediately fails the peer rather than queueing the action (which would have been fulfilled in line 494 above), but this code was originally developed in order to satisfy behavior of chunked transfers.

This behavior was evidenced when using the full Autobahn Test Suite.

libnanomsg of course never uses chunked transfers, but I was defensively coding against whether a browser (or user agent) might executively decide to chunk transfer from a javascript call that otherwise appeared like a single bulk transfer. At the time, there was no clear evidence whether browsers or user agents were granted the prerogative to make such a transformation based on app-level javascript.

(In the same vein, that is why PING/PONG existed here -- it was not evident whether browsers would have the authority to PING their peer, without having been requested to do so from client-side javascript.)

I don't have strong opinions right now, or even hard evidence to support one way or the other whether "chunked" could be chosen for us outside the will of either libnanomsg or a corresponding javascript implementation, but just wanted to share some design decisions.